### PR TITLE
Add SchemaBuilder

### PR DESCRIPTION
**Pull request by doanduyhai**
As promised, here is the **`SchemaBuilder`** to create/alter/drop table easily.

The code has all necessary Java doc and unit test coverage is 91% for classes and 96% for line coverage

I also added dependency to **`assertj`** with scope **`test`** so that assertions are more fluent than the classical **`assertTrue`**  and **`assertEquals`**

### CODE REVIEW

File: driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+    Alter(String tableName) {
+        validateNotEmpty(tableName, "Table name");
+        this.tableName = tableName;
+    }
+
+    /**
+     * Alter a column
+     * <p>
+     *     Please note that you cannot alter a column that is part of the primary key
```
Comment (olim7t) [inline driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java]: It works for me:

```
create table foo (s text primary key);
insert into foo (s) values ('bar');
alter table foo alter s type text;
```

File: driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        validateNotEmpty(columnName, "Column to be dropped");
+        return new StringBuilder(this.buildInternal())
+                .append(SPACE).append("DROP")
+                .append(SPACE).append(columnName).toString();
+    }
+
+    /**
+     * Rename a column
+     * <p>
+     *     Please note that you cannot rename a column that is part of the primary key
```
Comment (olim7t) [inline driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java]: Isn't it the other way around?

File: driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java (hunk 1 above)
Comment (doanduyhai) [inline driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java]: Exact, it's the other way

File: driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java (hunk 2 above)
Comment (doanduyhai) [inline driver-core/src/main/java/com/datastax/driver/core/schemabuilder/Alter.java]: Exact

**Comment by doanduyhai** [comment]
I'm going to push a new version with additional validation. Indeed we must check that a column is not added twice, once as simple column and once as static/clustering/partition column

**Comment by doanduyhai** [comment]
New version pushed

**Comment by olim7t** [comment]
Great contribution, thanks. A few additional remarks:

There are new table options in Cassandra 2.1:
- `index_interval` has been split into `min_index_interval` and `max_index_interval`
- `caching` is now a JSON-like map like `{"keys": "ALL", "rows_per_partition": 100}`. This could be handled with a new method `caching(Caching caching, int rowsPerPartition)`, and we would generate the 2.0 or 2.1 syntax depending on which method has been called.

Also, I think a free-form option method would be useful. Something like `withOption(String, String)`. This is in case new options appear in future C\* versions and the `SchemaBuilder` hasn't been updated yet.

One last thing: it would be nice to have a shortcut method for `new Create.Options.ClusteringOrder` as a static method on `SchemaBuilder`. Maybe call it `clusteringColumn`.

**Comment by doanduyhai** [comment]
Agree. I'll add the remarks

**Pull request closed without merging**